README.md

```diff
@@ -1,21 +1,94 @@
 # Kubevirt Velero Plugin
 
-Velero is the preferred solution for OpenShift for backup, disaster recovery
-and migration. Out-of-the-box it allows backing up VMs and DVs in a couple
-limited scenarios and requires jumping through several hoops.
+This repository contains Velero plugins. Thanks to this plugin, [Velero](https://velero.io/) can correctly backup and restore
+VirtualMachines, DataVolumess and other resources managed by [KubeVirt](https://kubevirt.io/) and 
+[CDI](https://github.com/kubevirt/containerized-data-importer/).
 
-The plugin will ensure full support as well as automation of the abovementioned
-hoop-jumping. ;-)
+For more information on Velero check https://velero.io/.
 
 This is still a work in progress, not intended for production use.
 
-## Building the plugins
+## Kinds of Plugins Included
+
+The plugin operates on following resources: DataVolume, PersistentVolumeClaim, Pod, VirtualMachine, VirtualMachineInstance.
+
+### **DVBackupItemAction** 
+A plugin that backs up the `PersistentVolumeClaim` and `DataVolume`
+ 
+Finds the PVC for DV and adds the `"cdi.kubevirt.io/storage.prePopulated" or "cdi.kubevirt.io/storage.populatedFor"` annotations
+### **VMBackupItemAction** 
+A plugin that backs up the `VirtualMachine`
+ 
+The plugin checks if a `VM` can be safely backed up and if the backup contains all required objects for the successful restore. 
+The plugin also returns the underlying `DataVolume` if a VM has `DataVolumeTemplate` and `virtualmachineinstances` as extra items to back up.
+
+### **VMIBackupItemAction** 
+A plugin that backs up the `VirtualMachineInstance`
+ 
+The plugin checks if a `VMI` can be safely backed up and if the backup contains all required objects for the successful restore.
+The plugin also returns the underlying VM volumes (`DataVolume` and `PersistentVolumeClaim`) and launcher `pod` as extra items to back up.
+
+### **VMRestoreItemAction**
+A plugin that restores the `VirtualMachine`
+ 
+Adds a `datavolumes` to list of restored items.
+
+### **VMIRestoreItemAction** 
+A plugin that restores the `VirtualMachineInstance`
+
+Skips the VMI if owned by a VM. The plugin also clears restricted labels, so the VMI is not rejected by kubevirt.  The restricted labels contain runtime information about the underlying KVM object.
+
+### **PodRestoreItemAction**
+A plugin that restores the virt-launcher `Pod`
```

Comment:
I'm surprised to see it. Restoring VMI is not enough?

Reply:
You are right, now when I checked the code again I see that the plugin just makes sure launcher pod is *NOT* restored:
"Launcher Pod should be unconditionally skipped"